src/static_array.cr

```diff
@@ -103,10 +190,19 @@ struct StaticArray(T, N)
     self
   end
 
+  # Transforms `self` into a slice
```

Comment:
Should be something like "Returns a slice that points to the elements of this static array. Changes made to the returned slice also affect this static array". Maybe use sample code like this:

```crystal
ary = StaticArray(Int32, 3).new(2)
ary # => [2, 2, 2]
slice = ary.to_slice #=> [2, 2, 2]
slice[0] = 3
ary # => [3, 2, 2]
```